### PR TITLE
Add so5extra/1.4.1

### PR DESCRIPTION
**Pull request by ngrodzitski**
This is header-only extensions to SObjectizer library (https://github.com/Stiffstream/so5extra).

Specify library name and version:  **so5extra/1.4.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/so5extra/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "so5extra"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "so5extra"
+        self.cpp_info.names["cmake_find_package"] = "sobjectizer"
+        self.cpp_info.names["cmake_find_package_multi"] = "sobjectizer"
+        self.cpp_info.components["so_5_extra"].name = "so5extra"
```
Comment (SpaceIm) [inline recipes/so5extra/all/conanfile.py:80]:
```suggestion
        self.cpp_info.components["so_5_extra"].names["cmake_find_package"] = "so5extra"
        self.cpp_info.components["so_5_extra"].names["cmake_find_package_multi"] = "so5extra"
```

File: recipes/so5extra/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["SO5EXTRA_INSTALL"] = True
+
+        self._cmake.configure(source_folder=os.path.join(self._source_subfolder, "dev", "so_5_extra"))
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
```
Comment (SpaceIm) [inline recipes/so5extra/all/conanfile.py:59]: Why not just copy header files? It's faster than calling CMake.

Comment (ngrodzitski) [reply recipes/so5extra/all/conanfile.py]: Just did it by example.
Though I would say that using CMake is more maintainable as some other day a project would require some generated files or select something based on parameters. This particular project does not use CMake heavily and has no plans to be intensely updated, so copy-approach is totally fine here.

Comment (ngrodzitski) [reply recipes/so5extra/all/conanfile.py]: Updated.

**Review by prince-chrismc — DISMISSED** [review]
TYpically we aim for smaller `test_package.cpp` but I am not even sure what's what 🙈

**Review by SSE4 — DISMISSED** [review]

File: recipes/so5extra/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+from conans import ConanFile, CMake, tools
```
Comment (SpaceIm) [inline recipes/so5extra/all/conanfile.py:1]:
```suggestion
from conans import ConanFile, tools
```

File: recipes/so5extra/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/Stiffstream/so5extra"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "A collection of various SObjectizer's extensions."
+    topics = ("concurrency", "actor-framework", "actors", "agents", "sobjectizer")
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    no_copy_source = True
+
+    _cmake = None
```
Comment (SpaceIm) [inline recipes/so5extra/all/conanfile.py:16]:
```suggestion
    settings = "compiler"
    no_copy_source = True
```

**Review by SpaceIm — COMMENTED** [review]
cleanup

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**